File: contains-duplicate-ii/contains-duplicate-ii.cpp
class Solution {
public:
    bool containsNearbyDuplicate(vector<int>& nums, int k) {
        unordered_map<int,int> pos;
        //update with latest occurence
        for(int it = 0 ; it<nums.size() ; it++){
            if(pos[nums[it]] == 0)
                pos[nums[it]]=it +1;
            else if( (it -  pos[nums[it]]) <k )
                return true;
            pos[nums[it]]=it +1;
            cout<<it<<" "<<pos[nums[it]]<<endl;
        }
        return false;
    }
};

File: contains-duplicate-ii/contains-duplicate-ii.java
class Solution {
    public boolean containsNearbyDuplicate(int[] nums, int k) {
        HashMap<Integer,Integer> hm=new HashMap<Integer,Integer>();
        for(int i=0;i<nums.length;i++)
        {
            if(!hm.containsKey(nums[i]))
            {
                hm.put(nums[i],i);
            }
            else
            {
                int diff=Math.abs(i-hm.get(nums[i]));
                if(diff<=k) return true;
            }
            hm.put(nums[i],i);
        }
        return false;
    }
}